src/shared.c

```diff
@@ -1026,6 +1026,14 @@ _Bool ValidateUserInfo(const char *s) {
 void SetUserInfo(char *s, const char *key, const char *value) {
 	char newi[MAX_USER_INFO_STRING], *v;
 
+	if (!value) {
+		return;
+	}
+
+	if (*value == '\0') {
+		return;
+	}
+
```

Comment:
Interesting. It must be a compiler thing. Because both sides of the || is evaluated right? How come when value is NULL,  *value == 0 doesn't crash?